opensrp-chw/src/kk/java/org/smartregister/chw/interactor/ChildHomeVisitInteractorFlv.java

```diff
@@ -1,4 +1,139 @@
+        } catch (Exception e) {
+            Timber.e(e);
+        }
+
+        bindEvents(serviceWrapperMap);
+
+        return actionList;
+    }
+
+    @Override
+    protected void bindEvents(Map<String, ServiceWrapper> serviceWrapperMap) throws BaseAncHomeVisitAction.ValidationException {
+        try {
+            evaluateBreastFeeding(serviceWrapperMap);
+            //evaluateECD();
+            evaluateExclusiveBreastFeeding(serviceWrapperMap);
+        } catch (BaseAncHomeVisitAction.ValidationException e) {
+            throw (e);
+        } catch (Exception e) {
+            Timber.e(e);
+        }
+
+    }
+
+    protected void evaluateBreastFeeding(Map<String, ServiceWrapper> serviceWrapperMap) throws Exception {
+
+        ServiceWrapper serviceWrapper = serviceWrapperMap.get("Essential care breastfeeding");
+        if (serviceWrapper == null) return;
+
+        Alert alert = serviceWrapper.getAlert();
+        if (alert == null || new LocalDate().isBefore(new LocalDate(alert.startDate()))) return;
+
+        final String serviceName = serviceWrapper.getName();
+
+        // Check if it is a dummy -5 weeks service that is there to re-set milestone to 0 before start 1 months recurring
+        if ("Essential care breastfeeding -5 weeks".equalsIgnoreCase(serviceName)) return;
+
+        // Get the very first breastfeeding visit
+        boolean firstBreastFeedingHappened;
+        List<Visit> breastFeedingServiceVisits = getVisitRepository().getVisits(memberObject.getBaseEntityId(), "Essential New Born Care: Breastfeeding");
+
+        firstBreastFeedingHappened = breastFeedingServiceVisits.size() > 0;
+
+        String title = getBreastfeedingServiceTittle(serviceWrapper.getName());
+
+        NewBornCareBreastfeedingHelper helper = new NewBornCareBreastfeedingHelper(context, alert, firstBreastFeedingHappened, serviceWrapper);
+        JSONObject jsonObject = getFormJson(KkConstants.KKJSON_FORM_CONSTANT.KKCHILD_HOME_VISIT.getChildHvBreastfeeding(), memberObject.getBaseEntityId());
+
+        BaseAncHomeVisitAction action = getBuilder(title)
+                .withHelper(helper)
+                .withDetails(details)
+                .withOptional(false)
+                .withBaseEntityID(memberObject.getBaseEntityId())
+                .withProcessingMode(BaseAncHomeVisitAction.ProcessingMode.SEPARATE)
+                .withPayloadType(BaseAncHomeVisitAction.PayloadType.SERVICE)
+                .withFormName(KkConstants.KKJSON_FORM_CONSTANT.KKCHILD_HOME_VISIT.getChildHvBreastfeeding())
+                .withPayloadDetails(serviceWrapper.getName())
+                .build();
+
+        actionList.put(title, action);
+
+    }
+
+    private String getBreastfeedingServiceTittle(String serviceName) {
+
+        String[] serviceNameSplit = serviceName.split(" ");
+        String period = serviceNameSplit[serviceNameSplit.length - 2];
```

Comment:
is this the same for services that have a single digit and the ones that have double digit?